src/ProcessMaker/Nayra/Bpmn/ProcessTrait.php

```diff
@@ -191,25 +287,29 @@ public function getTransitions(RepositoryFactoryInterface $factory)
             $transitions = array_merge($transitions, $gateway->getTransitions());
         }
         //Prepare the base events
-        $this->attachEvent(EventInterface::EVENT_EVENT_TRIGGERED, function (EventInterface $event) {
+        $this->attachEvent(EventInterface::EVENT_EVENT_TRIGGERED, function (EventInterface $event, TransitionInterface $transition, CollectionInterface $tokens) {
             if (!($event instanceof EndEventInterface)) return;
-            foreach ($this->getInstances() as $instance) {
-                if ($instance->getTokens()->count() !== 0) continue;
-                $this->notifyEvent(ProcessInterface::EVENT_PROCESS_COMPLETED, $this, $instance, $event);
-                $arguments = [$this, $instance, $event];
-                $bpmnEvents = $this->getBpmnEventClasses();
-                if (isset($bpmnEvents[ProcessInterface::EVENT_PROCESS_COMPLETED])) {
-                    $payload = new $bpmnEvents[ProcessInterface::EVENT_PROCESS_COMPLETED]($this, $arguments);
-                } else {
-                    $payload = ["object" => $this, "arguments" => $arguments];
-                }
-                $this->getDispatcher()->dispatch(ProcessInterface::EVENT_PROCESS_COMPLETED, $payload);
+            if ($tokens->count() === 0) return;
+            $instance = $tokens->item(0)->getInstance();
+            if ($instance->getTokens()->count() !== 0) return;
```

Comment:
Wouldn't it better to return an empty array []?, so we can avoid checking for nulls in the caller functions.